app/shell-config.js

```diff
@@ -8,10 +8,16 @@ var app = require("app");
 var fs = require("fs-extra");
 var path = require("path");
 var utils = require("./utils");
+var process = require("process");
+var os = require("os");
 
 var CONFIG_PATH = path.resolve(utils.convertWindowsPathToUnixPath(app.getPath("userData")), "shell-config.json");
 var config;
 
+if (!process.env["TMPDIR"] && !process.env["TMP"] && !process.env["TEMP"]) {
```

Comment:
I have no idea where it's picked up in the code. I assume somewhere in extensionManager.js or related file.

I pulled the code straight from Brackets-shell so I assume they wanted to set all 3 for a reason. I really don't know.